composer/utils/checkpoint.py

```diff
@@ -518,21 +510,35 @@ def read_data(self, plan: LoadPlan, planner: LoadPlanner):
                 # Ensure state exists
                 state_dict['state'] = state_dict.get('state', {})
 
+            # Only some ranks are meant to load checkpoint
+            expect_file = False
+            process_group = None
+            device_mesh = state.fsdp_device_mesh
+            if device_mesh is not None and device_mesh.ndim == 2:
+                expect_file = (device_mesh.get_local_rank(mesh_dim=0) == 0)
```

Comment:
add another clarification comment
```suggestion
                # mesh_dim=0 is the replication dimension, so only the devices on one of the replicas (we pick 0) need to save.
                expect_file = (device_mesh.get_local_rank(mesh_dim=0) == 0)
```